pkg/validator/validators.go

```diff
@@ -360,22 +391,53 @@ func validateMRCProvider(mrc *configv1alpha2.MeshRootCertificate) error {
 	return nil
 }
 
-// checkForExistingActiveMRC returns true if there are active MRCs in addition to the specified MRC
+// mrcWithIntentExist returns true if there are active MRCs in addition to the specified MRC
 // Returns false if there are no active MRCs or if the specified mrc will be the only active MRC
-func (kc *validator) checkForExistingActiveMRC(mrc *configv1alpha2.MeshRootCertificate) (bool, error) {
+func mrcWithIntentExist(mrcs []*configv1alpha2.MeshRootCertificate, mrc *configv1alpha2.MeshRootCertificate) bool {
+	for _, m := range mrcs {
+		if m.Spec.Intent == mrc.Spec.Intent && m.Name != mrc.Name {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (kc *validator) getCertsInUseMRCs() ([]*configv1alpha2.MeshRootCertificate, error) {
 	mrcs, err := kc.computeClient.ListMeshRootCertificates()
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	for _, m := range mrcs {
-		if m.Spec.Intent == v1alpha2.ActiveIntent && m.Name != mrc.Name {
-			log.Error().Msgf("cannot create MRC %s with intent active. An MRC with active intent already exists in the control plane namespace", getNamespacedMRC(mrc))
-			return true, nil
+	certsInUse := []*configv1alpha2.MeshRootCertificate{}
+	for _, mrc := range mrcs {
+		if mrc.Spec.Intent == v1alpha2.ActiveIntent || mrc.Spec.Intent == v1alpha2.PassiveIntent {
+			certsInUse = append(certsInUse, mrc)
+		}
+	}
+
+	return certsInUse, nil
+}
+
+func (kc *validator) testDesiredCertConfiguration(newMrcState *configv1alpha2.MeshRootCertificate) error {
```

Comment:
```suggestion
func (kc *validator) testDesiredCertConfiguration(newMRC *configv1alpha2.MeshRootCertificate) error {
```

This is an MRC, not a state, right? Also, consider a name like `simulateDesiredMRCConfiguration`